core/src/transaction.rs

```diff
@@ -74,53 +67,146 @@ impl rlp::Decodable for UnverifiedTransaction {
     }
 }
 
-impl rlp::Encodable for UnverifiedTransaction {
+impl TempTransaction {
+    /// Append object with a signature into RLP stream
+    fn rlp_append_sealed_transaction(&self, s: &mut RlpStream) {
+        s.begin_list(6);
+        s.append(&self.unsigned.seq);
+        s.append(&self.unsigned.fee);
+        s.append(&self.unsigned.network_id);
+        s.append(&self.unsigned.action);
+        s.append(&self.sig);
+        s.append(&self.signer_public);
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct Verified(TempTransaction);
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct Unverified(pub TempTransaction);
+
+impl rlp::Encodable for Verified {
     fn rlp_append(&self, s: &mut RlpStream) {
-        self.rlp_append_sealed_transaction(s)
+        self.0.rlp_append(s);
+    }
+}
+
+impl rlp::Decodable for Verified {
+    fn decode(d: &Rlp<'_>) -> Result<Self, DecoderError> {
+        let transaction = d.val_at(0)?;
+        Ok(Self(transaction))
+    }
+}
+
+impl Deref for Verified {
+    type Target = Transaction;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0.unsigned
     }
 }
 
-impl UnverifiedTransaction {
+impl From<Verified> for Unverified {
+    fn from(tx: Verified) -> Self {
+        Unverified(tx.0)
+    }
+}
+
+impl Verified {
+    /// Try to verify transaction and recover public.
+    pub fn try_new(tx: Unverified) -> Result<Self, Error> {
```

Comment:
Let's remove this method which converts `UnverifiedTransaction` into `VerifiedTransaction` without verification, and utilize `try_into()` instead.